src/layers/domain/core/cpm_system_id/v1.py

```diff
@@ -105,19 +111,29 @@ def latest_number(self):
 class ProductId(CpmSystemId):
     @classmethod
     def create(cls):
-        """No current_id needed, key is generated randomly."""
-        rng = random.Random(datetime.now().timestamp())
-        product_id = "-".join(
-            "".join(rng.choices(PRODUCT_ID_VALID_CHARS, k=PRODUCT_ID_PART_LENGTH))
-            for _ in range(PRODUCT_ID_NUMBER_OF_PARTS)
-        )
-        return cls(__root__=f"P.{product_id}")
+        existing_ids = cls.load_existing_ids()
+        while True:
+            """No current_id needed, key is generated randomly."""
+            rng = random.Random(datetime.now().timestamp())
+            product_id = "-".join(
+                "".join(rng.choices(PRODUCT_ID_VALID_CHARS, k=PRODUCT_ID_PART_LENGTH))
+                for _ in range(PRODUCT_ID_NUMBER_OF_PARTS)
+            )
+            if f"P.{product_id}" not in existing_ids:
+                return cls(__root__=f"P.{product_id}")
```

Comment:
to minimise the change, perhaps instead could just change the `return` line of the function to:

```
        if f"P.{product_id}" in cls.load_existing_ids():
            return cls.create()
        return cls(__root__=f"P.{product_id}")
```

Reply:
I did it this way so that the file is called once. In your suggestion the file would get called on each recursive flow. Seems

Reply:
i'd cache the load under the hood to avoid loading the file twice, so e.g.

```python
@cache
def _load_existing_ids():
   # load the ids

...

class CantRememberTheNameOfTheClass:
   ...

   @classmethod
   def load_existing_ids(cls):
       return _load_existing_ids()

   ...
```